app/frontend/src/features/search/FilterDialog.test.tsx

```diff
@@ -0,0 +1,149 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { HOSTING_STATUS } from "features/constants";
+import { hostingStatusLabels } from "features/profile/constants";
+import {
+  APPLY_FILTER,
+  LAST_ACTIVE,
+  LAST_WEEK,
+  LOCATION,
+  NUM_GUESTS,
+} from "features/search/constants";
+import { HostingStatus } from "pb/api_pb";
+import { Route } from "react-router-dom";
+import hookWrapper from "test/hookWrapper";
+import { server } from "test/restMock";
+
+import FilterDialog from "./FilterDialog";
+
+const renderDialog = (
+  mutableParams: URLSearchParams,
+  setSearchQuery?: (value: string) => void
+) => {
+  render(
+    <>
+      <FilterDialog
+        isOpen={true}
+        onClose={() => {}}
+        searchParams={mutableParams}
+      />
+      <Route
+        path="*"
+        render={({ location }) => {
+          setSearchQuery?.(location.search);
+          return null;
+        }}
+      />
+    </>,
+    {
+      wrapper: hookWrapper,
+    }
+  );
+};
+
+describe("FilterDialog", () => {
```

Comment:
@darrenvong need your testing wisdom again, I'm getting `An update to FilterDialog inside a test was not wrapped in act` warnings (I thought act is not necessary when using `userEvent`?)

Reply:
After some digging (since there's nothing obviously wrong in your test), it looks like this is a quirk with `react-hook-form` that others have hit as well, namely the validation happens async which falls outside of the synchronous `act` boundary that `userEvent` would have wrapped around things like `type` - https://github.com/testing-library/user-event/issues/128#issuecomment-720119616

So it looks like the recommended thing to do by the linked comment is to "wait a bit" for the async operations from `react-hook-form` to complete properly. Wrapping the assertions inside an `await waitFor(() => { // assertions }` seem to be enough